packages/mermaid/src/diagrams/quadrant-chart/quadrantBuilder.ts

```diff
@@ -0,0 +1,385 @@
+// @ts-ignore: TODO Fix ts errors
+import { scaleLinear } from 'd3';
+
+export type QuadrantPointInputType = { x: number; y: number; text: string };
+
+export type TextVerticalPos = 'left' | 'center' | 'right';
+export type TextHorizontalPos = 'top' | 'middle' | 'bottom';
+
+export type QuadrantTextType = {
+  text: string;
+  fill: string;
+  x: number;
+  y: number;
+  verticalPos: TextVerticalPos;
+  horizontalPos: TextHorizontalPos;
+  fontSize: number;
+  rotation: number;
+};
+
+export type QuadrantPointType = {
+  x: number;
+  y: number;
+  fill: string;
+  radius: number;
+  text: QuadrantTextType;
+};
+
+export type QuadrantQuadrantsType = {
+  text: QuadrantTextType;
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+  fill: string;
+};
+
+export type QuadrantBuildType = {
+  points: QuadrantPointType[];
+  quadrants: QuadrantQuadrantsType[];
+  axisLabels: QuadrantTextType[];
+};
+
+export class QuadrantBuilder {
+  private _quadrant1Text = '';
+  private _quadrant2Text = '';
+  private _quadrant3Text = '';
+  private _quadrant4Text = '';
+  private _xAxisLeftText = '';
+  private _xAxisRightText = '';
+  private _yAxisBottomText = '';
+  private _yAxisTopText = '';
+  private _totalHeight = 500;
+  private _totalWidth = 500;
+  public quadrantPadding = 5;
+  public xAxisLabelPadding = 5;
+  public yAxisLabelPadding = 5;
+  public xAxisLabelFontSize = 16;
+  public yAxisLabelFontSize = 16;
+  public quadrantLabelFontSize = 16;
+  public quadrantTextTopPadding = 5;
+  public pointTextPadding = 5;
+  public pointLabelFontSize = 12;
+  public pointRadius = 5;
+  public points: QuadrantPointInputType[] = [];
+  public xAxisPosition = 'top';
+  public yAxisPosition = 'left';
+  public quadrant1Fill = '#8bc2f3';
+  public quadrant2Fill = '#faebd7';
+  public quadrant3Fill = '#00ffff';
+  public quadrant4Fill = '#f0ffff';
+  public quadrant1TextFill = '#93690e';
+  public quadrant2TextFill = '#8644ff';
+  public quadrant3TextFill = '#e3004d';
+  public quadrant4TextFill = '#000000';
+  public pointFill = '#60B19C';
+  public pointTextFill = '#0000ff';
+  public xAxisTextFill = '#000000';
+  public yAxisTextFill = '#000000';
+  public showXAxis = true;
+  public showYAxis = true;
```

Comment:
Use a config object of type `QuadrantChartConfig` instead of exposing each config individually.
Then you can use the default values defined in defaultConfig.ts instead of duplicating the values here. 
Also, assignment would be easier.

Instead of
```ts
    quadrantBuilder.quadrantPadding = quadrantChartConfig.quadrantPadding;
    quadrantBuilder.xAxisLabelPadding = quadrantChartConfig.xAxisLabelPadding;
    quadrantBuilder.yAxisLabelPadding = quadrantChartConfig.yAxisLabelPadding;
    quadrantBuilder.xAxisLabelFontSize = quadrantChartConfig.xAxisLabelFontSize;
    quadrantBuilder.yAxisLabelFontSize = quadrantChartConfig.yAxisLabelFontSize;
    quadrantBuilder.quadrantLabelFontSize = quadrantChartConfig.quadrantLabelFontSize;
    quadrantBuilder.quadrantTextTopPadding = quadrantChartConfig.quadrantTextTopPadding;
    quadrantBuilder.pointTextPadding = quadrantChartConfig.pointTextPadding;
    quadrantBuilder.pointLabelFontSize = quadrantChartConfig.pointLabelFontSize;
    quadrantBuilder.pointRadius = quadrantChartConfig.pointRadius;
    quadrantBuilder.xAxisPosition = quadrantChartConfig.xAxisPosition;
    quadrantBuilder.yAxisPosition = quadrantChartConfig.yAxisPosition;
```
Call
```
quadrantBuilder.setConfig(quadrantChartConfig);
```

and inside setConfig 
```ts
public setConfig(updatedConfig: QuadrantChartConfig) {
    this.config = { ...this.config, ...updatedConfig }
}
```

Reply:
done